java/src/jmri/jmrix/cmri/serial/networkdriver/configurexml/ConnectionConfigXml.java

```diff
@@ -52,33 +52,35 @@ protected void getInstance(Object object) {
     // though it would be good to fix it if you're working in this area
     protected void extendElement(Element e) {
         SerialTrafficController tc = ((CMRISystemConnectionMemo)adapter.getSystemConnectionMemo()).getTrafficController();
-        SerialNode node = (SerialNode) tc.getNode(0);
-        int index = 1;
-        while (node != null) {
-            // add node as an element
-            Element n = new Element("node");
-            n.setAttribute("name", "" + node.getNodeAddress());
-            e.addContent(n);
-            // add parameters to the node as needed
-            n.addContent(makeParameter("nodetype", "" + node.getNodeType()));
-            n.addContent(makeParameter("bitspercard", "" + node.getNumBitsPerCard()));
-            n.addContent(makeParameter("transmissiondelay", "" + node.getTransmissionDelay()));
-            n.addContent(makeParameter("num2lsearchlights", "" + node.getNum2LSearchLights()));
-            n.addContent(makeParameter("pulsewidth", "" + node.getPulseWidth()));
-            String value = "";
-            for (int i = 0; i < node.getLocSearchLightBits().length; i++) {
-                value = value + Integer.toHexString(node.getLocSearchLightBits()[i] & 0xF);
+        if (tc != null) {
```

Comment:
I _think_ that if this code runs with tc==null, you're going to store without the node definitions, thereby losing the node definitions.  Is that really what you want?